pkg/apply/task/apply_task.go

```diff
@@ -128,22 +142,48 @@ func (a *ApplyTask) Start(taskContext *taskrunner.TaskContext) {
 	}()
 }
 
-func (a *ApplyTask) sendTaskResult(taskContext *taskrunner.TaskContext, err error) {
-	taskContext.TaskChannel() <- taskrunner.TaskResult{
-		Err: err,
+func newApplyOptions(eventChannel chan event.Event, strategy common.DryRunStrategy, factory util.Factory) (applyOptions, error) {
+	discovery, err := factory.ToDiscoveryClient()
+	if err != nil {
+		return nil, err
+	}
+	dynamic, err := factory.DynamicClient()
+	if err != nil {
+		return nil, err
 	}
-}
 
-func (a *ApplyTask) setApplyOptionsFields(eventChannel chan event.Event) {
-	if ao, ok := a.ApplyOptions.(*apply.ApplyOptions); ok {
-		ao.DryRun = a.DryRunStrategy.ClientDryRun()
-		ao.ServerDryRun = a.DryRunStrategy.ServerDryRun()
-		adapter := &KubectlPrinterAdapter{
+	emptyString := ""
+	return &apply.ApplyOptions{
```

Comment:
Not technically necessary, but maybe useful for documentation: we may want to set the pre-processor and post-processor hooks to nil here.

Reply:
I thought about it, but I think we want to either set all properties, or just set the ones that are needed. Since the pre-processor and post-processor hooks are nil by default, I decided not to set them here.